skyve-war/src/main/java/modules/admin/UserLoginRecord/jobs/DifferentCountryLoginNotificationJob.java

```diff
@@ -0,0 +1,71 @@
+package modules.admin.UserLoginRecord.jobs;
+
+import java.util.List;
+
+import org.skyve.impl.util.UtilImpl;
+import org.skyve.job.Job;
+import org.skyve.util.CommunicationUtil;
+
+import modules.admin.ModulesUtil;
+import modules.admin.domain.Contact;
+import modules.admin.domain.UserLoginRecord;
+
+/**
+ * This job is used to send an email to a user notifying them that their account was accessed from a different country. It is called
+ * when the system notices a change in the user's country from their last log in.
+ * The job uses a {@code UserLoginRecord} bean parameter that's passed when the scheduler calls the job.
+ */
+public class DifferentCountryLoginNotificationJob extends Job {
+
+	private static final String SUCCESS_MESSAGE = "Successfully sent email warning of unusual login activity to %s";
+
+	public static final String JOB_NAME = "jDifferentCountryLoginNotification";
+
+	private static final String UNUSUAL_LOGIN_ACTIVITY_EMAIL_DEFAULT_SUBJECT = "Security Alert: Unusual Login Activity Detected";
+
+	private static final String UNUSUAL_LOGIN_EMAIL_BODY = String.format("Dear {name},\n\n"
+			+ "We have detected a login attempt to your account from a new location: {country} with IP address: {ipAddress}. "
+			+ "If this was you, there's no need to take further action.\n\n"
+			+ "However, if you do not recognize this activity, we strongly recommend that you change your password immediately to secure your account. "
+			+ "You can do this by logging into your account and navigating to <strong>Admin -> Password</strong>.\n\n"
+			+ "For your safety, please do not share your password with anyone.\n\n"
+			+ "If you have any questions or need assistance, please contact our support team at " +  UtilImpl.SUPPORT_EMAIL_ADDRESS + ".\n\n"
+			+ "Best regards,\n"
+			+ "The Security Team", Contact.namePropertyName,UserLoginRecord.countryPropertyName, UserLoginRecord.ipAddressPropertyName);
```

Comment:
This string format is not doing anything - you are not using the static attribute values parsed. 
Also, adding the support email address statically here will mean that a change in support email address from `Configuration`/`Startup` will not be reflected in the email contents.

Check out my similar email implementation in `SendPasswordChangeNotificationJob`:

Use this email body I re-wrote:
```
private static final String EMAIL_BODY = "Dear {"
			+ Contact.namePropertyName
			+ "},<br/><br/>"
			+ "We have detected a login attempt to your account from a new location: {"
			+ UserLoginRecord.countryPropertyName
			+ "} with IP address: {"
			+ UserLoginRecord.ipAddressPropertyName
			+ "}. If this was you, there's no need to take further action.<br/><br/>"
			+ "However, if you do not recognize this activity, we strongly recommend that you change your password immediately to secure your account."
			+ "You can do this by logging into your account and navigating to <strong>Admin -> Password</strong>.<br/><br/>"
			+ "For your safety, please do not share your password with anyone.<br/><br/>"
			+ "If you have any questions or need assistance, please contact our support team at {"
			+ Startup.environmentSupportEmailPropertyName
			+ "}<br/><br/>"
			+ "Best regards,<br/>"
			+ "The Security Team";
```
Get the startup configuration like this:
```
// Get startup configuration
		Startup startup = Startup.newInstance();
		if (startup.getEnvironmentSupportEmail() == null) {
			String warningMessage = "There is no environment support email specified. Failed to send different country login notification.";
			log.add(warningMessage);
			Util.LOGGER.warning(warningMessage);
			return;
		}
```
and parse the startup bean to the `sendFailSafeSystemCommunication` method:
```
CommunicationUtil.sendFailSafeSystemCommunication(COMMUNICATION_DESCRIPTION,
						UNUSUAL_LOGIN_ACTIVITY_EMAIL_DEFAULT_SUBJECT,
						UNUSUAL_LOGIN_EMAIL_BODY,
						CommunicationUtil.ResponseMode.EXPLICIT, null, contact, loginRecord, startup);
```

Reply:
Do you think having the support email required is ideal in this case? Practically means the users can not get the warning unless it is set

Reply:
Had a discussion with Ben about this - the environment support email should never be null - it set by default in Foundry. If a user receives a support email and can't contact anyone about it (support email is null), it may be useless.